cocotb/share/lib/gpi_log/gpi_logging.c

```diff
@@ -142,7 +141,7 @@ void gpi_log(const char *name, long level, const char *pathname, const char *fun
         return;
     }
 
-    if (level < local_level)
+    if (level < (long)local_level)
```

Comment:
The better type for both `level` and `local_level` is `enum gpi_log_levels`. `local_level` is already of that type, so please change the `level` parameter to be of that type as well. (Most callers of this function come through the `LOG_*` functions in `include/gpi_logging.h` which already uses the enum constants.)